File: lib/features/dictionary/domain/entities/app_context.dart
// lib/features/dictionary/domain/entities/app_context.dart
enum AppContext {
  general,
  business,
  technology,
  travel,
  foodAndDrink,
  health,
  academic,
  socialCasual;

  String get label => switch (this) {
        AppContext.general => 'General',
        AppContext.business => 'Business',
        AppContext.technology => 'Technology',
        AppContext.travel => 'Travel',
        AppContext.foodAndDrink => 'Food & Drink',
        AppContext.health => 'Health',
        AppContext.academic => 'Academic',
        AppContext.socialCasual => 'Social/Casual',
      };

  String get emoji => switch (this) {
        AppContext.general => '🌐',
        AppContext.business => '💼',
        AppContext.technology => '💻',
        AppContext.travel => '✈️',
        AppContext.foodAndDrink => '🍜',
        AppContext.health => '🏥',
        AppContext.academic => '📚',
        AppContext.socialCasual => '💬',
      };
}

File: apps/web/src/lib/appContext.ts
export type AppContext =
  | "general"
  | "business"
  | "technology"
  | "travel"
  | "foodAndDrink"
  | "health"
  | "academic"
  | "socialCasual";

export const APP_CONTEXTS: AppContext[] = [
  "general",
  "business",
  "technology",
  "travel",
  "foodAndDrink",
  "health",
  "academic",
  "socialCasual",
];

// Mirrors lib/features/dictionary/domain/entities/app_context.dart's
// AppContextX.label/.emoji — kept in sync manually (no shared-types
// package between the Flutter and web apps).
export const APP_CONTEXT_LABELS: Record<AppContext, string> = {
  general: "General",
  business: "Business",
  technology: "Technology",
  travel: "Travel",
  foodAndDrink: "Food & Drink",
  health: "Health",
  academic: "Academic",
  socialCasual: "Social/Casual",
};

export const APP_CONTEXT_EMOJI: Record<AppContext, string> = {
  general: "🌐",
  business: "💼",
  technology: "💻",
  travel: "✈️",
  foodAndDrink: "🍜",
  health: "🏥",
  academic: "📚",
  socialCasual: "💬",
};
